### PR TITLE
Report Coverage to Code Climate

### PR DESCRIPTION
**Pull request by EGiataganas**
Did the basic installation for code climate. Added the gem file and then
added test reporter on the very first line of the spec_helper.

### CODE REVIEW

File: spec/spec_helper.rb [hunk 1]
```diff
@@ -1,3 +1,6 @@
+require "codeclimate-test-reporter"
+CodeClimate::TestReporter.start
+
 require 'simplecov'
 require 'simplecov-rcov'
 SimpleCov.formatter = SimpleCov::Formatter::RcovFormatter
```
Comment (threedaymonk) [inline spec/spec_helper.rb]: I think that we want to run simplecov only if the CodeClimate environment variable is *not* set, so this should be something like:

```ruby
if ENV.key?('CODE_CLIMATE_WHATEVER')
  require "codeclimate-test-reporter"
  CodeClimate::TestReporter.start
else
  require 'simplecov'
  require 'simplecov-rcov'
  SimpleCov.formatter = SimpleCov::Formatter::RcovFormatter
end
```

**Comment by dan-palmer** [comment]
Nice to see you using `fixup` :star2:

**Comment by threedaymonk** [comment]
There's an element of uncertainty here in that we're not sure whether it will actually report coverage until it's merged into the master branch, because CodeClimate will only report coverage on the master branch, and it won't be reporting on the same commit until then.

Still, I think this is fine to merge.

File: Gemfile.lock [hunk 2]
```diff
@@ -350,3 +353,6 @@ DEPENDENCIES
   unicorn
   virtus
   zendesk_api
+
+BUNDLED WITH
+   1.10.6
```
Comment (threedaymonk) [inline Gemfile.lock]: @dan-palmer @SteveMarshall Shall we bite this bullet?

**Comment by dan-palmer** [comment]
Can confirm this as working :ship:

**Pull request merged**